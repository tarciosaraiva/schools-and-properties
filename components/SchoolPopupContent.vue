<template>
  <div class="popup-info container">
    <div class="row">
      <div class="column">
        <p class="school-name">
          <strong>{{ school.schoolName }}</strong>
        </p>
      </div>
    </div>
    <div class="row">
      <span class="column school-stat-label">Overall rating</span>
      <span class="column school-stat-value"><em>{{ school.primaryOverallScore }}</em></span>
    </div>
    <div class="row">
      <span class="column school-stat-label">English rating</span>
      <span class="column school-stat-value"><em>{{ school.primaryEnglishScore }}</em></span>
    </div>
    <div class="row">
      <span class="column school-stat-label">Maths rating</span>
      <span class="column school-stat-value"><em>{{ school.primaryMathsScore }}</em></span>
    </div>
    <div class="row">
      <span class="column school-stat-label">Enrolments</span>
      <span class="column school-stat-value"><em>{{ school.primaryEnrolments }}</em></span>
    </div>
    <div class="row">
      <span class="column school-stat-label">Phone</span>
      <span class="column school-stat-value"><em>{{ school.phoneNumber }}</em></span>
    </div>
  </div>
</template>

<script lang="ts">
import Vue from 'vue'

export default Vue.extend({
  props: {
    school: Object,
  },
})
</script>
