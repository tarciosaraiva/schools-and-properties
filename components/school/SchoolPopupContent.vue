<template>
  <div class="popup-info container">
    <p class="school-name">
      {{ school.schoolName }}
      <br>
      <small>{{ school.educationSector }}, {{ school.schoolType }} - {{ school.phoneNumber }}</small>
    </p>
    <p v-if="isPrimary && hasPrimaryRating">Primary ratings</p>
    <dl v-if="isPrimary && hasPrimaryRating" class="school-stats">
      <dt class="label">Overall rating</dt>
      <dd class="value"><em>{{ school.primaryOverallScore }}</em></dd>
      <dt class="label">English rating</dt>
      <dd class="value"><em>{{ school.primaryEnglishScore }}</em></dd>
      <dt class="label">Maths rating</dt>
      <dd class="value"><em>{{ school.primaryMathsScore }}</em></dd>
      <dt class="label">Enrolments</dt>
      <dd class="value"><em>{{ school.primaryEnrolments }}</em></dd>
    </dl>
    <br v-if="isPrimary && !hasPrimaryRating" />
    <p v-if="isPrimary && !hasPrimaryRating">Check school rating at
      <a href="https://www.bettereducation.com.au/school/Primary/vic/vic_primary_school_rating.aspx" target="_blank">Better Education</a>.
    </p>
    <p v-if="isSecondary && hasSecondaryRating">Secondary ratings</p>
    <dl v-if="isSecondary && hasSecondaryRating" class="school-stats">
      <dt class="label">Overall rating</dt>
      <dd class="value"><em>{{ school.secondaryOverallScore }}</em></dd>
      <dt class="label">English rating</dt>
      <dd class="value"><em>{{ school.secondaryEnglishScore }}</em></dd>
      <dt class="label">Maths rating</dt>
      <dd class="value"><em>{{ school.secondaryMathsScore }}</em></dd>
      <dt class="label">Enrolments</dt>
      <dd class="value"><em>{{ school.secondaryEnrolments }}</em></dd>
    </dl>
    <br v-if="isSecondary && !hasSecondaryRating" />
    <p v-if="isSecondary && !hasSecondaryRating">Check school rating at
      <a href="https://www.bettereducation.com.au/school/secondary/vic/vic_secondary_school_rating.aspx" target="_blank">Better Education</a>.
    </p>
  </div>
</template>

<script lang="ts">
import Vue from 'vue'

export default Vue.extend({
  props: {
    school: {
      type: Object,
      default: () => ({})
    }
  },
  computed: {
    isPrimary () {
      const primSchool = this.school
      return primSchool.schoolType === 'Primary' || primSchool.schoolType === 'Pri/Sec'
    },
    isSecondary () {
      const secSchool = this.school
      return secSchool.schoolType === 'Secondary' || secSchool.schoolType === 'Pri/Sec'
    },
    hasPrimaryRating () {
      const primSchool = this.school
      return primSchool.primaryOverallScore !== '#N/A'
    },
    hasSecondaryRating () {
      const secSchool = this.school
      return secSchool.secondaryOverallScore !== '#N/A'
    }
  }
})
</script>
