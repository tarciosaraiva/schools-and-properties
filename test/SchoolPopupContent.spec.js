import { shallowMount } from '@vue/test-utils'
import SchoolPopupContent from '@/components/SchoolPopupContent.vue'

describe('SchoolPopupContent', () => {
  const propsData = {
    school: {
      schoolName: 'school',
      overallScore: 95,
      englishScore: 4,
      mathsScore: 4,
      enrolments: 500,
      phoneNumber: '03 9999 8888',
    }
  }

  test('is a Vue instance', () => {
    const wrapper = shallowMount(SchoolPopupContent, { propsData })
    expect(wrapper.vm).toBeTruthy()
  })

  test('renders correctly with props', () => {
    const wrapper = shallowMount(SchoolPopupContent, { propsData })
    expect(wrapper).toMatchSnapshot()
  })
})
