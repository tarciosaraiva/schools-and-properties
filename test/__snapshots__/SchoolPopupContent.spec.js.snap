// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SchoolPopupContent renders correctly with props 1`] = `
<div class="school-info container">
  <div class="row">
    <div class="column">
      <p class="school-name">
        <strong>school</strong>
      </p>
    </div>
  </div>
  <div class="row">
    <span class="column school-stat-label">Overall rating</span>
    <span class="column school-stat-value">
      <em>95</em>
    </span>
  </div>
  <div class="row">
    <span class="column school-stat-label">English rating</span>
    <span class="column school-stat-value">
      <em>4</em>
    </span>
  </div>
  <div class="row">
    <span class="column school-stat-label">Maths rating</span>
    <span class="column school-stat-value">
      <em>4</em>
    </span>
  </div>
  <div class="row">
    <span class="column school-stat-label">Enrolments</span>
    <span class="column school-stat-value">
      <em>500</em>
    </span>
  </div>
  <div class="row">
    <span class="column school-stat-label">Phone</span>
    <span class="column school-stat-value">
      <em>03 9999 8888</em>
    </span>
  </div>
</div>
`;
